timemachine/fe/single_topology.py

```diff
@@ -1197,55 +1192,53 @@ def _parameterize_host_guest_nonbonded(self, lamb, host_nonbonded):
             if membership == 0:  # core atom
                 a_idx = self.c_to_a[idx]
                 b_idx = self.c_to_b[idx]
-                plane = 0
-                offset = 0
+
                 # interpolate charges when in common-core
                 q = (1 - lamb) * guest_a_q[a_idx] + lamb * guest_b_q[b_idx]
                 sig = (1 - lamb) * guest_a_lj[a_idx, 0] + lamb * guest_b_lj[b_idx, 0]
                 eps = (1 - lamb) * guest_a_lj[a_idx, 1] + lamb * guest_b_lj[b_idx, 1]
+
+                # fixed at w = 0
+                w_coords = 0.0
+
             elif membership == 1:  # dummy_A
                 a_idx = self.c_to_a[idx]
-                plane = 0
-                offset = 1
                 q = guest_a_q[a_idx]
                 sig = guest_a_lj[a_idx, 0]
                 eps = guest_a_lj[a_idx, 1]
+
+                # w: 0 -> cutoff
+                w_coords = lamb * cutoff
+
             elif membership == 2:  # dummy_B
                 b_idx = self.c_to_b[idx]
-                plane = -1
-                offset = 1
                 q = guest_b_q[b_idx]
                 sig = guest_b_lj[b_idx, 0]
                 eps = guest_b_lj[b_idx, 1]
+
+                # w: -cutoff -> 0
+                w_coords = (lamb - 1.0) * cutoff
```

Comment:
nit: took a few seconds to parse the comments as specifying boundary conditions for interpolation (possible style collision with function description `f: x -> y`).

May expand the wording a bit to `# interpolate from w = 0 to w = cutoff`, or `# w = -cutoff @ lamb=0, w = 0 @ lamb=1` or similar, or perhaps avoid need for comment using `w_coords = lerp(lamb, -cutoff, 0)` or similar...

Reply:
Good call, I like the suggestion to use `lerp`. Updated in faede63